### DISCUSSION TITLE
e107 Version 1.0.4 : L'objet qui est vide mais  contient =?utf-8?Q??=

### DISCUSSION THREAD

**Discussion by Pascal570** [e107inc/e107#5353] [General]
Bonjour,

Je suis nouveau inscrit. Malheureusement je ne parle que le français. Je ne sais pas si je suis au bon endroit pour demander votre aide.
 Je possède un site e107 depuis 12/02/2005. Je suis toujours en v. 1.0.4.
Je me suis fait aider par un ami, depuis un mois pour passer le site de PHP 5  en PHP 8.1.30. 
J'ai un soucis avec les mails
Option  SMTP 
L'objet des mails est toujours vide mais qui contient à la place: =?utf-8?Q??=
Comment résoudre ce problème?
Je vous remercie 

Pascal
https://www.l3fr.org

**Comment by Jimmi08** [comment]
Salut,
Essaies-tu de mettre à jour la version 1.0.4 vers PHP 8 ? Ton site fonctionnait-il sous PHP 7.4 ?
Vérifie l'encodage / le charset dans la base de données. Sans voir les modifications du code original, il est difficile de dire ce qui ne va pas.
La version 1.0.4 ne fonctionnera pas sous PHP 8 sans modifications, donc tu as dû en faire quelques-unes.
Peux-tu confirmer si tu essaies de mettre à jour la version 1.0.4 vers PHP 8 ou vers la nouvelle version d’e107 (2.3.3) ? Quelle est ta situation ?

JM

**Comment by Pascal570** [comment]
Salut JM,
Le site est toujours en e107 v1.0.4 et tournait depuis longtemps en  PHP  5.6.40 jusqu'au 17 septembre dernier.
Le 17 septembre nous avons basculé le site (qui est toujours en e107 v1.0.4) vers Version PHP 8.1.30.
Effectivement  beaucoup de fichiers affichaient  "deprecated" ont été corrigés. 
Actuellement le site tourne bien. Ce sont tous les mails émis depuis e107 qui posent problème. J'ai testé l'envoi de mails directement depuis le serveur l'objet fonctionne très bien.
Depuis  /e107_admin/mailout.php?prefs en option **j'ai opté smtp** au lieu de php 

En PHP les messages comportaient le même soucis + au autre : les messages étaient retournés
Le corps des messages contenait  genre de chose
Content-Type: text/plain; charset = "utf-8"
Content-Transfer-Encoding: 8bit


Pascal

**Reply by Jimmi08** [reply]
Salut, 
as-tu résolu le problème de connexion à la base de données ? Les fonctions `mysql_` ne fonctionneront pas avec PHP 8, cela entraînera une erreur fatale, pas seulement des avertissements "deprecated".

Concernant les mails retournés – c'est dû à la valeur FROM dans les en-têtes de ton mail. De nos jours, le premier filtre anti-spam vérifie que l'en-tête FROM doit être identique au domaine de l'adresse e-mail.

Tu peux trouver cela dans handlers/mail.php, cherche` $mail->From = $send_from;.` Cela peut provenir du formulaire lui-même, du champ 'replyto_email' ou du champ 'siteadminemail'. 
Tu peux envoyer un e-mail de test à une adresse Gmail et vérifier cela directement là-bas.  

De nos jours, la politique DKIM, SPF et DMARC est le seul moyen de s'assurer que tes e-mails sont reçus en toute sécurité.

Parfois, c'est un vrai cauchemar. Pour SMTP, assure-toi que ton 'siteadminemail' provient du même domaine.

Le problème avec le contenu – si tu n'as pas mis à jour PHPMailer, il aura des problèmes avec PHP 8. Cela ressemble à un problème d'analyse... Ce code (Content-Transfer-Encoding) provient directement de PHPMailer.

Je ne suis pas sûr que tout soit bien traduit. Si tu as des difficultés à comprendre, n'hésite pas à redemander.
